graph_search.py

```diff
@@ -0,0 +1,115 @@
+class GraphSearch:
+  def dfs(self, target, graph, starting_vertex):
+    info = {
+      "target": target,
+      "found": False,
+      "visited": [],
+      "checks_whether_vertex_is_target": 0
+    }
+
+    def traverse(vertex):
+      info["visited"].append(vertex)
+      info["checks_whether_vertex_is_target"] += 1
+      if (vertex == target):
+        info["found"] = True
+        return info
+
+      for v in graph[vertex]:
+        if v not in info["visited"]:
```

Comment:
Given that `info["visited"]` is a list, what's the running time implication on (i) this step, (ii) the entire process?  What would be a data structure that gives you an improved running time?